### ISSUE TITLE
Bug: Calling Ch() in looping select is incorrect

### ISSUE THREAD

**Issue by Nokel81** [lmittmann/chanio#1]
A common use can for channels is in a `select` statement, especially when that is in a `for` statement. However, the general way to do this `case o, open := <-scanner.Ch()` is incorrect since a new channel is created everytime which the races to read the input from the previously created channel.

**Comment by lmittmann** [comment]
Yes, with current API the `Ch()` method should be called only once, as it creates a new channel on every call (which isn't really documented). In your scenario you could write something like this:

```go
ch := scanner.Ch()
for {
    select {
    case o, open := <-ch:
        // do something
    }
}
```

It might be more intuitive to expose a single channel with `Ch()`. However, this would require changing the filter `FilterFunc` logic.

**Comment by Nokel81** [comment]
Yes that is what I ended up doing. In doing so to confirm what I thought was going on. I think that either this should be documented in the readme or the channel should be added the structure definition and then use that if not nil.
